datadog_checks_dev/datadog_checks/dev/docker.py

```diff
@@ -22,4 +32,219 @@ def get_container_ip(container_id_or_name):
+        if self.service_name:
+            self.command.append(self.service_name)
+
+    def __call__(self):
+        return run_command(self.command, check=True)
+
+
+class ComposeFileDown(LazyFunction):
+    def __init__(self, compose_file, check=True):
+        self.compose_file = compose_file
+        self.check = check
+        self.command = ['docker-compose', '-f', self.compose_file, 'down']
+
+    def __call__(self):
+        return run_command(self.command, check=self.check)
+
+
+class CheckEndpoints(LazyFunction):
+    def __init__(self, endpoints, timeout=1, attempts=60, wait=1):
+        self.endpoints = endpoints
+        self.timeout = timeout
+        self.attempts = attempts
+        self.wait = wait
+
+    def __call__(self):
+        last_endpoint = ''
+        last_error = ''
+
+        for _ in range(self.attempts):
+            for endpoint in self.endpoints:
+                last_endpoint = endpoint
+                try:
+                    request = urlopen(endpoint, timeout=self.timeout)
+                except URLError as e:
+                    last_error = str(e)
+                    break
+                else:
+                    status_code = request.getcode()
+                    if 400 <= status_code < 600:
+                        last_error = 'status {}'.format(status_code)
+                        break
+            else:
+                break
+
+            time.sleep(self.wait)
+        else:
+            raise RetryError(
+                'Endpoint: {}\n'
+                'Error: {}'.format(
+                    last_endpoint,
+                    last_error
+                )
+            )
+
+
+class CheckComposeFileLogs(LazyFunction):
+    def __init__(self, compose_file, patterns, matches=1, attempts=60, wait=1):
+        self.compose_file = compose_file
+        self.attempts = attempts
+        self.wait = wait
+        self.command = ['docker-compose', '-f', self.compose_file, 'logs']
+
+        if matches == 'all':
+            self.matches = len(patterns)
+        else:
+            self.matches = matches
+
+        self.patterns = [
+            re.compile(pattern, re.M) if isinstance(pattern, string_types) else pattern
+            for pattern in patterns
+        ]
+
+    def __call__(self):
+        log_output = ''
+
+        for _ in range(self.attempts):
+            result = run_command(self.command, capture=True, check=True)
+            log_output = result.stdout
+            matches = 0
+
+            for pattern in self.patterns:
+                if pattern.search(log_output):
+                    matches += 1
+
+            if matches >= self.matches:
+                break
+
+            time.sleep(self.wait)
+        else:
+            raise RetryError(
+                'Command: {}\n'
+                'Captured Output: {}'.format(
+                    self.command,
+                    log_output
+                )
+            )
+
+
+class CheckContainerLogs(LazyFunction):
```

Comment:
Can we maybe merge this with `CheckComposeFileLogs` by adding some logic ? I don't think the duplication is really needed just to change `self.command`

Reply:
@gzussa beat me to it !!